spikeinterface/preprocessing/remove_artifacts.py

```diff
@@ -89,25 +95,37 @@ def __init__(self, recording, list_triggers, ms_before=0.5, ms_after=3.0, mode='
         fit_sample_range = fit_sample_interval * 2 + 1
         fit_samples = np.arange(0, fit_sample_range, fit_sample_interval)
 
+        artefact = None
+
         BasePreprocessor.__init__(self, recording)
         for seg_index, parent_segment in enumerate(recording._recording_segments):
             triggers = list_triggers[seg_index]
-            rec_segment = RemoveArtifactsRecordingSegment(parent_segment, triggers, pad, mode, fit_samples)
+            rec_segment = RemoveArtifactsRecordingSegment(parent_segment, triggers, pad, mode, fit_samples, artefact)
             self.add_recording_segment(rec_segment)
 
+        if mode in ['median', 'average']:
+            sorting = NumpySorting.from_times_labels(list_triggers, [np.ones(len(i)) for i in list_triggers], recording.get_sampling_frequency())
+            waveforms_params = {'ms_before' : ms_before, 'ms_after' : ms_after}
+            w = extract_waveforms(recording, sorting, None, mode='memory', **waveforms_params, return_scaled=False)
+            artefact = w.get_all_templates(mode=mode)[0]
+
+            for seg_index, parent_segment in enumerate(recording._recording_segments):
+                self._recording_segments[seg_index].artefact = artefact
+
```

Comment:
Can you move this before instantiating the segment, so you can directly pass the `artifact`?

Reply:
Indeed, this has been solved